server/src/instant/db/transaction.clj

```diff
@@ -356,6 +356,8 @@
                            (:deep-merge-triple results)
                            (:retract-triple results)))
             _  (triple-model/validate-required! conn attrs app-id eid+attr-ids)
```

Comment:
I do worry about this. I am not 100% sure how our average perf looks for transactions (A quick look shows P50 at around [20ms](https://ui.honeycomb.io/instantdb/environments/prod/datasets/instant-server/result/m9ZniL9kJn4)), so even 100ms extra would be meaningful increase in latency. 

I know you mentioned it's going to be really hard to guarantee that all required attributes are satisfied purely with the data we already have. Depending on how the optimization goes on this query, it may be worth revisiting.

Reply:
Okay, I figured it out. It was missing `app-id = '...'` when querying triples, I guess without it indexes couldn’t kick in because app-id is part of all of them.

Re: speed, I’m now getting consistent 110-120ms, most of which seem to be my own latency, because I get the same numbers on simple queries to prod, too. So we should be good here

Reply:
Niice! Okay, let's shiip!